src/atomate2/abinit/jobs/response.py

```diff
@@ -0,0 +1,430 @@
+    else:
+        perts = [{"idir": 1}, {"idir": 2}, {"idir": 3}]
+
+    return perts
+
+
+@job
+def generate_dte_perts(
+    gsinput: AbinitInput,
+    # TODO: or gsinput via prev_outputs?
+    skip_permutations: bool | None = False,
+    phonon_pert: bool | None = False,
+    ixc: int | None = None,
+) -> list[dict[str, int]]:
+    """
+    Generate the perturbations for the DTE calculations.
+
+    Parameters
+    ----------
+    gsinput : an |AbinitInput| representing a ground state calculation,
+        likely the SCF performed to get the WFK.
+    skip_permutations: Since the current version of abinit always performs
+        all the permutations of the perturbations, even if only one is asked,
+        if True avoids the creation of inputs that will produce duplicated outputs.
+    phonon_pert: is True also the phonon perturbations will be considered.
+        Default False.
+    ixc: Value of ixc variable. Used to overwrite the default value read
+        from pseudos.
+    """
+    # Call Abinit to get the list of irreducible perturbations
+    gsinput = gsinput.deepcopy()
+    gsinput.pop_vars(["autoparal", "npfft"])
+    perts = gsinput.abiget_irred_dteperts(
+        phonon_pert=phonon_pert, ixc=ixc
+    )  # TODO: quid manager?
+
+    if skip_permutations:
+        perts_to_skip: list = []
+        reduced_perts = []
+        for pert in perts:
+            p = (
+                (pert.i1pert, pert.i1dir),
+                (pert.i2pert, pert.i2dir),
+                (pert.i3pert, pert.i3dir),
+            )
+            if p not in perts_to_skip:
+                reduced_perts.append(pert)
+                perts_to_skip.extend(itertools.permutations(p))
+
+        perts = reduced_perts
+
+    return perts
+
+
+@job
+def run_ddk_rf(
+    perturbations: list[dict],
+    ddk_maker: DdkMaker | None = None,
+    prev_outputs: list[str] | None = None,
+) -> Flow:
+    """
+    Run the DDK calculations.
+
+    Parameters
+    ----------
+    perturbations : a list of dict with the direction of the perturbation
+        under the Abipy format.
+    prev_outputs : a list of previous output directories
+    gsinput : an |AbinitInput| representing a ground state calculation,
+        likely the SCF performed to get the WFK.
+    ddk_maker : Maker to create a job with a DDK ABINIT calculation.
+    """
+    if ddk_maker is None:
+        ddk_maker = DdkMaker()
+
+    ddk_jobs = []
+    outputs: dict[str, list] = {"dirs": []}
+
+    for ipert, pert in enumerate(perturbations):
+        ddk_job = ddk_maker.make(
+            perturbation=pert,
+            prev_outputs=prev_outputs,
+            # structure=structure, removed because of
+            #   factory_prev_inputs_kwargs already used
+        )
+        ddk_job.append_name(f"{ipert+1}/{len(perturbations)}")
+        ddk_job = update_user_abinit_settings(
+            ddk_job, {"tolwfr": 1e-1}
+        )  # VT TO REMOVE, ONLY FOR TESTING
```

Comment:
Sure. But this is linked to another issue I pointed out in the PR description. Basically, modifying the tol_wfr directly in the response script is the only way I found to control the tolerance of the DDK or DDE... this is clearly not ideal... I tried using powerups in the main DFPT flow but it does not work because the DdkMaker only appears after it has replaced the run_ddk job...

Reply:
After some digging I think that the fact that the powerup are not working in this case might be solved by changing the `ddk_maker` definition. 
In the definition use
```python
ddk_maker: DdkMaker = field(default_factory=DdkMaker)
```

and remove the lines above 
```python
    if ddk_maker is None:
        ddk_maker = DdkMaker()
```

The same for the other functions that generate dde and dte jobs.
This should make the powerups also update those jobs. This may still not be a general solution and I will probably open an issue in jobflow about the problem, but it should hopefully solve the problem here. Can you make this test and tell me if it actually works?